src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/filter/AuditFilter.java

```diff
@@ -0,0 +1,112 @@
+
+import com.amazon.opendistroforelasticsearch.security.auditlog.AuditConfig;
+import com.amazon.opendistroforelasticsearch.security.auditlog.AuditLog;
+import com.amazon.opendistroforelasticsearch.security.auditlog.impl.AuditCategory;
+import com.amazon.opendistroforelasticsearch.security.support.WildcardMatcher;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.transport.TransportRequest;
+
+import java.util.EnumSet;
+import java.util.List;
+
+public class AuditFilter {
+
+    protected static final Logger log = LogManager.getLogger(AuditFilter.class);
+
+    private static boolean checkIgnoredAuditUser(final List<String> ignoredAuditUsers, final String effectiveUser) {
+        if (ignoredAuditUsers.size() > 0 && WildcardMatcher.matchAny(ignoredAuditUsers, effectiveUser)) {
+            return false;
+        }
+        return true;
+    }
+
+    private static boolean checkIgnoredAuditRequest(final List<String> ignoredAuditRequests, final RestRequest request) {
+        if (request != null && ignoredAuditRequests.size() > 0
+                && (WildcardMatcher.matchAny(ignoredAuditRequests, request.path()))) {
+            return false;
+        }
+        return true;
+    }
+
+    private static boolean checkIgnoredAuditRequest(final List<String> ignoredAuditRequests, final TransportRequest request) {
+        if (request != null && ignoredAuditRequests.size() > 0
+                && (WildcardMatcher.matchAny(ignoredAuditRequests, request.getClass().getSimpleName()))) {
+            return false;
+        }
+        return true;
+    }
+
+    private static boolean checkDisabledCategories(final EnumSet<AuditCategory> disabledCategories, final AuditCategory category) {
+        return !disabledCategories.contains(category);
+    }
+
+    private static boolean checkInternal(String action) {
+        return action == null ||
+                (!action.startsWith("internal:")
+                        && !action.startsWith("cluster:monitor")
+                        && !action.startsWith("indices:monitor"));
+    }
+
+    private static boolean checkComplianceOrigin(final AuditLog.Origin origin, final String effectiveUser, final AuditCategory category) {
+        if (origin == AuditLog.Origin.LOCAL && effectiveUser == null && category != AuditCategory.COMPLIANCE_EXTERNAL_CONFIG) {
+            return false;
+        }
+        return true;
+    }
+
+    private static boolean checkComplianceUser(final List<String> ignoredComplianceUsers, final String effectiveUser, final EnumSet<AuditCategory> complianceCategories, final AuditCategory category) {
+        if (complianceCategories.contains(category)) {
+            if (ignoredComplianceUsers.size() > 0 && effectiveUser != null
+                    && WildcardMatcher.matchAny(ignoredComplianceUsers, effectiveUser)) {
+                return false;
+            }
+        }
```

Comment:
Log.trace(..) messages are missing from the original code. Similarly, check other methods that are supposed to write trace messages.